### PR TITLE
Restruct and skip clear imp err

### PR DESCRIPTION
**Pull request by gabemontero**

### CODE REVIEW

**Comment by gabemontero** [comment]
terraform flake

/retest

**Comment by gabemontero** [comment]
ok got to run and e2e-aws-operator passed

**Comment by gabemontero** [comment]
@xiuwang found an existing bug along with her readme comment that this PR missed (pertains to our hashing our progressing==true on errors to alert ops per clusteroperator spec)

commit pushed

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -1178,28 +592,27 @@ func (h *Handler) Handle(event v1.Event) error {
...
-			logrus.Debugf("CRDUPDATE bad spec validation update")
+			logrus.Printf("CRDUPDATE bad spec validation update")
 			// only retry on error updating the Config; do not return
 			// the error from SpecValidation which denotes a bad config
 			return h.crdwrapper.UpdateStatus(cfg)
 		}
 		// if a bad config was corrected, update and return
 		if existingValidStatus != cfg.Condition(v1.ConfigurationValid).Status {
-			logrus.Debugf("CRDUPDATE spec corrected")
+			logrus.Printf("CRDUPDATE spec corrected")
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply pkg/stub/handler.go]: @bparees and I discussed this on slack .... short answer yes

we'll revisit the precise greppable pattern and make the phrasing more legible to non-maintainers

File: pkg/stub/config.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+		changeInContent := false
+		if skip != cfg.Status.SkippedImagestreams[i] {
+			changeInContent = true
+			logrus.Printf("SkippedImagestreams changed in content from %s to %s", cfg.Status.SkippedImagestreams[i], skip)
+			importErrors := h.clearStreamFromImportError(skip, cfg.Condition(v1.ImportImageErrorsExist), cfg)
+			if importErrors != nil {
+				clearImageImportErrors = true
+			}
+		}
+		return changeInContent, changeInContent, clearImageImportErrors
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply pkg/stub/config.go]: yep you are correct

File: pkg/stub/handler.go [hunk 3]
```diff
@@ -147,15 +144,16 @@ func (h *Handler) prepWatchEvent(kind, name string, annotations map[string]strin
...
 	// if by some slim chance sample watch events come in before the first
-	// Config event get the file locations and skipped lists
+	// Config event return error for retry
 	if len(h.imagestreamFile) == 0 || len(h.templateFile) == 0 {
-		for _, arch := range cfg.Spec.Architectures {
-			dir := h.GetBaseDir(arch, cfg)
-			files, _ := h.Filefinder.List(dir)
-			h.processFiles(dir, files, cfg)
-		}
+		return nil, "", false, fmt.Errorf("samples file locations not build yet when event for %s/%s came in", kind, name)
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply pkg/stub/handler.go]: yep

**Comment by gabemontero** [comment]
updates for your comments pushed @bparees

**Comment by bparees** [comment]
squash and lgtm.
/approve

**Pull request merged**